spec/index.bs

```diff
@@ -1637,6 +1637,22 @@ interface GPU {
             <!-- If we add ways to make invalid adapter requests (aside from those
                  that violate IDL rules), specify that they reject the promise. -->
         </div>
+
+    : <dfn>getPreferredCanvasFormat()</dfn>
+    ::
+        Returns an optimal {{GPUTextureFormat}} to use for {{GPUCanvasContext}}s on this system.
+
+        <div algorithm="GPU.getPreferredCanvasFormat">
+            **Called on:** {{GPU}} this.
+
+            **Returns:** {{GPUTextureFormat}}
+
+            <div class=content-timeline>
+                1. Return an optimal {{GPUTextureFormat}} to use when calling
+                    {{GPUCanvasContext/configure()}} on a {{GPUCanvasContext}} on this system. Must
+                    be one of the [=supported context formats=].
+            </div>
+        </div>
```

Comment:
I realized while reviewing the meeting minutes (to check we had consensus), we should explicitly scope this to just rgba8unorm and bgra8unorm. And be more explicit about what it's intended to be optimal for.

```suggestion
        Returns an optimal {{GPUTextureFormat}} to use when calling
        {{GPUCanvasContext/configure()}} on a {{GPUCanvasContext}}, for displaying
        8-bit depth, standard dynamic range content to the screen: either
        {{GPUTextureFormat/"rgba8unorm"}} or {{GPUTextureFormat/"bgra8unorm"}}.

        Canvases which are not displayed to the screen may or may not benefit from
        using this format.

        <div algorithm="GPU.getPreferredCanvasFormat">
            **Called on:** {{GPU}} this.

            **Returns:** {{GPUTextureFormat}}

            <div class=content-timeline>
                1. Out of {{GPUTextureFormat/"rgba8unorm"}} and {{GPUTextureFormat/"bgra8unorm"}},
                    return the format which is optimal for displaying WebGPU canvases on this system.
            </div>
        </div>
```

Reply:
Which raises the point: In theory we could optimize `new OffscreenCanvas`es and canvas elements not added to the DOM to avoid a copy-to-bgra (on IOSurface) - cc @austinEng

Reply:
Actually, this makes me wonder: In what's scenarios would you use an `OffscreenCanvas` with WebGPU and not eventually transfer it to a DOM element?

`OffscreenCanvas` primarily exists because the majority of Canvas APIs had to have a Canvas-like object to host the context, so an `OffscreenCanvas` was the only way to use the API at all in a Worker. WebGPU doesn't have that problem, so the only purpose for using it with an `OffscreenCanvas` is either:
  
  - Presenting to a Canvas from a worker
  - Generating a screenshot/image file without adding an element to the DOM.

In the former case a copy-to-bgra would still happen, in the latter who cares, because it's probably not a realtime operation?

Reply:
OffscreenCanvas is also necessary for interop (WebGPU -> other stuff like 2d, webgl, video encoders) on workers.